core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -726,7 +726,7 @@ class Partition(val topicPartition: TopicPartition,
           s"with partition epoch ${partitionState.partitionEpoch}, high watermark ${leaderLog.highWatermark}, " +
           s"ISR ${isr.mkString("[", ",", "]")}, adding replicas ${addingReplicas.mkString("[", ",", "]")} and " +
           s"removing replicas ${removingReplicas.mkString("[", ",", "]")} ${if (isUnderMinIsr) "(under-min-isr)" else ""}. " +
-          s"Previous leader epoch was $leaderEpoch.")
+          s"Previous leader $leaderReplicaIdOpt and previous leader epoch was $leaderEpoch.")
```

Comment:
since it is an optional, shall we choose a sane default when it is missing, such as "unknown"

Reply:
It will printed out as `None` when not defined. I thought that it is good enough but I can change it if you feel strong about this.

Reply:
No, it's ok. No need to change.